Update Google logo icon

ecgan opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Link to P2: pe2C5g-kK-p2#comment-160

In this PR, as per the request in the P2 above, we update the Google logo icon from the "Google shopping tag" icon to generic colorful "G" icon.

In the `google.js` file, I use `img` element with `WC_ASSET_URL` to point to the svg file, so that we don't need to have duplicated svg code in the JS file.

Additional technical changes:

- I simplified the `ProductIcon` component by using the `size` prop and removing custom CSS for svg element. The `size` prop applies to both svg and img elements.

### Screenshots

In "Recommended extensions" card:

<img width="899" alt="image" src="https://user-images.githubusercontent.com/417342/186970198-58033dbf-7791-4a85-b5bb-327c5a47befe.png">

In "Installed marketing extensions" card:

<img width="892" alt="image" src="https://user-images.githubusercontent.com/417342/186970354-9362ce6e-7c73-4a99-b95b-c18130f733ca.png">

### How to test the changes in this Pull Request:

1. Go to the Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`
2. You should see the "Recommended extensions" card. Google logo icon should be as per the above screenshot.
3. Install the Google Listings and Ads plugin.
4. In the "Installed marketing extensions" card, Google logo icon should be as per the above screenshot.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

ecgan committed:
Update Google logo icon.

ecgan committed:
Update Google logo icon to use img element with svg file.

ecgan committed:
Simplify ProductIcon CSS with size prop.

This applies to svg and image icons, not just svg.

ecgan committed:
Add changelog.

chihsuan reviewed (dismissed):
@ecgan Thanks for updating the logo. Tested well and looks good. 🚢

ecgan committed:
Set style for Google logo icon.

We set white background to override the gray backgroun in ProductIcon.

Padding is used to make it look visually similar size with other icons.

ecgan commented:
@chihsuan , thank you for the review. May I know how do we get the PR merged to trunk branch? It seems like I am not authorized to merge this PR. Can I be authorized? 😄 

<img width="625" alt="image" src="https://user-images.githubusercontent.com/417342/186885722-f049f71e-58ae-4449-be41-9f9cf933a932.png">

Also, I just made a small styling commit [`5d1df3a` (#34471)](https://github.com/woocommerce/woocommerce/pull/34471/commits/5d1df3a9211adb0bc076c6a08ab9c72d513f490b) for the Google icon, as per request in pcTzPl-QI-p2#comment-1575. Let me know if you have comment on it. 🙏

ecgan committed:
Update style for Google logo.

chihsuan reviewed (approved):
Looks good!
